### DISCUSSION TITLE
Fast C API in V8

### DISCUSSION THREAD

**Discussion by uasan** [uNetworking/uWebSockets.js#424] [General]
I don't understand much in V8 )
but you complained about the slow call of Node.js, maybe this solves this problem:
> The motivation for this work comes from the observation that the JavaScript bindings between V8 and the embedder turns out to be the bottleneck for performance critical functions that get optimized by TurboFan. The original implementation was done having Blink and the WebGL/WebGPU bindings in mind, but this work is supposed to serve the needs of other embedders and their performance sensitive calls, possibly calling into hardware APIs.

https://v8.dev/blog/v8-release-87#unsafe-fast-js-calls

**Comment by ghost** [comment]
This can be useful but this is not the bottleneck. There's nothing wrong with V8 - I like V8. But Node.js is not V8. Node.js takes V8 and wraps it in an immense amount of derailing bloat that severely limits the potentials of native modules.
